lumen/ai/assistant.py

```diff
@@ -302,6 +306,12 @@ async def _choose_agent(self, messages: list | str, agents: list[Agent], return_
             return agent_names[0]
         self._current_agent.object = "## **Current Agent**: [Lumen.ai](https://lumen.holoviz.org/)"
         agent_model = self._create_agent_model(agent_names)
+
+        for agent in agents:
+            if isinstance(agent, CustomAnalysisAgent):
+                analyses = "\n".join((f"`{analysis.__name__}`: {analysis.__doc__.strip()}" for analysis in agent.analyses))
+                agent.__doc__ = f"Available analyses include:\n{analyses}\nSelect this agent to perform one of these analyses."
+
         system = render_template(
```

Comment:
Outputs:
```python

- `TableListAgent`: List all of the available tables or datasets inventory. Not useful if the user requests a specific table.

- `hvPlotAgent`: Generates a plot of the data given a user prompt. If the user asks to plot, visualize or render the data this is your best best.

- `TableAgent`: Displays a single table / dataset. Does not discuss.

- `ChatAgent`: Chats and provides info about high level data related topics, e.g. what datasets are available, the columns of the data or statistics about the data, and continuing the conversation. Is capable of providing suggestions to get started or comment on interesting tidbits. If data is available, it can also talk about the data itself.

- `ChatDetailsAgent`: Responsible for chatting and providing info about details about the table values, e.g. what should be noted about the data values, valuable, applicable insights about the data, and continuing the conversation. Does not provide overviews; only details, meaning, relationships, and trends of the data.

- `CustomAnalysisAgent`: Available analyses include: `CumulativeNinoAnalysis`: Counts the number of consecutive months of any Nino/neutral event. `TotalNinoAnalysis`: Counts the total number of Nino/neutral events. Select this agent to perform one of these analyses.
```

Given: 

```python

class CumulativeNinoAnalysis(CustomAnalysis):

    """
    Counts the number of consecutive months of any Nino/neutral event.
    """

    def __call__(self, pipeline):
        return hvPlotView(pipeline=pipeline, x="year", y="cumulative")


class TotalNinoAnalysis(CustomAnalysis):
    
    """
    Counts the total number of Nino/neutral events.
    """

    def __call__(self, pipeline):
        return hvPlotView(pipeline=pipeline, x="year", y="ntotal")


```